main.rb

```diff
@@ -0,0 +1,570 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # My each method definition
+
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    arr = self
+    if arr.is_a? Array
+      0.upto(arr.length - 1) do |index|
+        yield (arr[index])
+      end
+    elsif arr.is_a? Range
+      arr.step(1) do |value|
+        yield value
+      end
+    end
+    arr
+  end
+
+  # my_each_with_index method definition
+  def my_each_with_index
+    return to_enum :my_each unless block_given?
+
+    arr = self
+    0.upto(arr.length - 1) do |ii|
+      yield arr[ii], ii
+    end
+    arr
+  end
+
+  # my_select method definition
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    arr = self
+    back = []
+    arr.my_each { |x| back << x if yield x }
+    back
+  end
+
+  def inc_on_true(cont, cond)
+    cont += 1 if cond
+    cont
+  end
+
+  def param_reg(par, arr_i, t_reg, t_patt, who)
+    # who is false for my_all method call
+    # who is true for my_none method call
+    return [t_reg, t_patt] if par.nil?
+
+    if par.class == Regexp && (!par.match(arr_i).nil? ^ who)
+      t_reg += 1
+
+    elsif conditions(par.class != Regexp, par.class == Class, true)
+      t_patt = inc_on_true(t_patt, arr_i.is_a?(Module.const_get(par.to_s)) ^ who)
+
+    elsif conditions(par.class != Regexp, par.class != Class, true)
+      t_patt = inc_on_true(t_patt, (arr_i == par) ^ who)
+    end
+    [t_reg, t_patt]
+  end
+
+  # my_all? method definition
+  def my_all?(param = nil)
+    arr = self
+    any_h = { t_b_e: 0, t_e: 0, t_rxp: 0, t_p: 0, who: false }
+
+    0.upto(arr.length - 1) do |i|
+      if param.nil?
+        if block_given?
+          any_h[:t_b_e] += 1 if yield arr[i]
+        elsif !arr[i].nil? && arr[i] != false
+          any_h[:t_e] += 1
+        end
+      end
+      any_h[:t_rxp], any_h[:t_p] = param_reg(param, arr[i], any_h[:t_rxp], any_h[:t_p], false)
+    end
+    any_element_true(arr.length, any_h[:t_b_e], any_h[:t_e], any_h[:t_rxp], any_h[:t_p])
+
+    # if we reach this step and all elements are thruty, we exit true.
+  end
+
+  def no_block_count(element_check, block, arr_i, param)
+    element_check += 1 if (!arr_i.nil? || arr_i != false) && !block && param.nil?
+    element_check
+  end
+
+  # my_any? method definition
+  def my_any?(param = nil)
+    arr = self
+    any_h = { true_block_elements: 0, true_elements: 0, t_rxp: 0, true_pattern: 0, who: false }
+
+    0.upto(arr.length - 1) do |i|
+      if block_given?
+        any_h[:true_block_elements] += 1 if yield arr[i]
+      end
+      any_h[:t_rxp], any_h[:true_pattern] = param_reg(param, arr[i], any_h[:t_rxp], any_h[:true_pattern], false)
+      any_h[:true_block_elements] = no_block_count(any_h[:true_block_elements], block_given?, arr[i], param)
+    end
+    any_h[:true_block_elements] >= 1 || any_h[:t_rxp] >= 1 || any_h[:true_pattern] >= 1
+  end
```

Comment:
For `my_any?`, handle the case when no block or argument is given returns true if at least one of the collection is not false or nil

- Built-in method
`[nil, false, nil, false].any?` returns `false`

- Your method
`[nil, false, nil, false].my_any?` returns `true`

- Check this [document](https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-any-3F) for detailed information on `any?` method.

Reply:
Thanks for this feedback. this method is now fixed on my end.
irb(main):448:0> [nil, false, nil, false].any?
=> false
irb(main):449:0> [nil, false, nil, false].my_any?
=> false
irb(main):450:0>